homeassistant/components/publibike/sensor.py

```diff
@@ -0,0 +1,112 @@
+"""Support for PubliBike Public API for bike sharing in Switzerland."""
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import BATTERY_LIMIT, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the PubliBike sensors."""
+    station = hass.data[DOMAIN][config_entry.entry_id]["station"]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+    battery_limit = hass.data[DOMAIN][config_entry.entry_id][BATTERY_LIMIT]
+    entities = [
+        EBikeSensor(station, coordinator, battery_limit),
+        BikeSensor(station, coordinator),
+    ]
+
+    async_add_entities(entities)
+
+
+class PubliBikeSensor(CoordinatorEntity, SensorEntity):
+    """Representation of a PubliBike sensor."""
+
+    def __init__(self, station, coordinator):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.station = station
+        self.coordinator = coordinator
+        self.attributes = {}
+        self.bike_type = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self.station.name} - {self.bike_type.capitalize()}s"
```

Comment:
It less code to just handle that using a shorthand attribute in the child classes

```suggestion
```